### DISCUSSION TITLE
a few questions

### DISCUSSION THREAD

**Discussion by d99000706** [opendatacam/opendatacam#537] [Q&A]
We have a traffic project that may be able to make use of ODC but have some questions.
[Here is a video in a very early stage](https://physimetricsinc-my.sharepoint.com/:v:/g/personal/chris_physimetrics_com/ETXcymyFPcxEiSaTQ3Myg10BkW0510J2qeKLad4sxImvsw?e=TbdhEc).

What video resolution(s) does ODC support?

Our project is different in at least a couple ways.

- our camera view is not conveniently looking at traffic head on but an an arbitrary angle
- instead of counting how many vehicles cross a boundary line, we want to know how many entered the intersection from a certain road and of those how many turned left, right or went straight

How difficult do you think it would be to modify your open source code to do this?
We are experienced C++ programmers but limited in linux and js.

How are you getting 10 fps (or better) performance from yolo?  We are using yolov3-608 in the video above and are only getting about 2.5 fps on an old GeForce GTX1050.  The inference time is the bottleneck and takes about 0.4 seconds per frame.
Do you have a recommendation for a particular gpu for yolov3-608?
Do you have any other recommendations for improved performance?

**Comment by maddust** [comment]
It looks like you are trying to do a turning movement count .  I'm using opendatacam with custom portable cameras with telescoping poles and custom  nerual network model to do the same .

My advise looking at your video is to place the camera closer to the intersection (as high as posible).  Then with the help of the tracker you can set lines at each movement only counting with the direction of traffic.

Another solution is to set lines at each approach and then pairing the I'd of vehicles from the origin to the last line passed (post processing). Although I found this method to be less accurate due to vehicles switching IDs if they get blocked or mis tracked.

As for the resolution ODC accepts pretty much any resolution. But for best performance you should input resolution as close or equal as your neural network resolution  since darknet needs to resizes every frame of the video to the model resolution.anyways to do the inference

I'm getting 325fps on a rtx 3070 with a resolution of 640x352 (same as my model). My model is trained in yolov4 tiny 3L . And 12fps on a Jetson nano.

Here a Example how to set the lines. For big intersections i use 2 cameras  on opposite corners 

https://www.youtube.com/embed/LiXFulmTEhY
